modules/@apostrophecms/attachment/index.js

```diff
@@ -402,16 +401,21 @@ module.exports = {
             extensions: accepted.join(req.t('apostrophe:listJoiner'))
           }));
         }
+
+        if (options.attachmentId && await self.apos.attachment.db.findOne({ _id: options.attachmentId })) {
+          throw self.apos.error('invalid', 'duplicate');
```

Comment:
This could hit a race condition and imposes an unnecessary delay, try/catch the insert and check for `self.apos.doc.isUniqueError(err)` to see if mongodb threw a unique key error or something else.